sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -89,441 +59,41 @@ private SwaggerMethodParser getMethodParser(Method method) {
         return interfaceParser.getMethodParser(method);
     }
 
-    /**
-     * Send the provided request asynchronously, applying any request policies provided to the HttpClient instance.
-     *
-     * @param request the HTTP request to send
-     * @param contextData the context
-     * @return a {@link Mono} that emits HttpResponse asynchronously
-     */
-    public Mono<HttpResponse> send(HttpRequest request, Context contextData) {
-        return httpPipeline.send(request, contextData);
-    }
-
     @Override
     public Object invoke(Object proxy, final Method method, Object[] args) {
         RestProxyUtils.validateResumeOperationIsNotPresent(method);
 
         try {
             final SwaggerMethodParser methodParser = getMethodParser(method);
-            final HttpRequest request = createHttpRequest(methodParser, args);
+
+            HttpRequest request;
+            boolean isReactive = isReactive(methodParser.getReturnType());
+            if (isReactive) {
+                request = asyncRestProxy.createHttpRequest(methodParser, args);
+            } else {
+                request = syncRestProxy.createHttpRequest(methodParser, args);
+            }
```

Comment:
should this get into respective `SyncProxy`/`AsyncProxy` ?
I.e. I'd expect only one `isReactive` switch in this class and respective implementations carry on.